app/assets/javascripts/settings.js.erb

```diff
@@ -27,8 +27,18 @@ See doc/COPYRIGHT.rdoc for more details.
 
 ++#%>
 
-jQuery(document).ready(function($) {
-  $('#settings_session_ttl_enabled').on('change', function(){
-    $('#settings_session_ttl_container').toggle($(this).is(':checked'));
-  }).trigger('change');
-});
+(function($) {
+  $(function() {
+
+    $('#settings_session_ttl_enabled').on('change', function(){
+      $('#settings_session_ttl_container').toggle($(this).is(':checked'));
+    }).trigger('change');
+
+    $('#settings_repositories_automatic_managed_repos').change(function() {
+      var checked = this.checked;
+      $('#settings_repositories_automatic_managed_vendor').prop('disabled', function() {
```

Comment:
Hmm, I think this is rather a question of changing the disabled style of the select, and the rest is matter of taste - I prefer having a separated switch rather.